src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,105 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.util.stream.Stream;
+
+/**
+ * Represents a Parser to parse date and time.
+ */
+public class Parser {
+
+    /**
+     * Constructs an instance of Parser.
+     */
+    public Parser() {
+        // Empty
+    }
+
+    /**
+     * Parses a string representation of a date into LocalDate.
+     * Expects input to be formatted as "YYYY-MM-DD" (fixed to "-" seperator).
+     * TODO: need to catch exceptions
+     *
+     * @param dateStr String representation of a date.
+     * @return Date in a LocalDate instance.
+     */
+    private LocalDate parseDate(String dateStr) {
+        return LocalDate.parse(dateStr);
+    }
+
+    /**
+     * Parses a string representation of a date into LocalDate.
+     * Expects input to be formatted as "YYYY-MM-DD" but seperator is arbitrary.
+     * Arbitrary seperator needs to be specified.
+     * TODO: need to catch exceptions
+     *
+     * @param dateStr String representation of a date.
+     * @param seperator Seperator used to separate the day, month, year.
```

Comment:
should be spelt as "sepArator", not "sepErator".

Reply:
Thank you for your feedback! I will make changes accordingly.